apps/nestjs-indexer/src/indexer/tip-service.indexer.ts

```diff
@@ -0,0 +1,278 @@
+      blockTimestamp: new Date(Number(blockTimestamp.seconds) * 1000),
+      sender: senderAddress,
+      nostrRecipient,
+      tokenAddress,
+      amount: Number(amount),
+    };
+
+    await this.tipService.createDeposit(data);
+  }
+
+  private async handleTipClaimEvent(
+    header: starknet.IBlockHeader,
+    event: starknet.IEvent,
+    transaction: starknet.ITransaction,
+  ) {
+    const {
+      blockNumber,
+      blockHash: blockHashFelt,
+      timestamp: blockTimestamp,
+    } = header;
+
+    const blockHash = validateAndParseAddress(
+      `0x${FieldElement.toBigInt(blockHashFelt).toString(16)}`,
+    ) as ContractAddress;
+
+    const transactionHashFelt = transaction.meta.hash;
+    const transactionHash = validateAndParseAddress(
+      `0x${FieldElement.toBigInt(transactionHashFelt).toString(16)}`,
+    ) as ContractAddress;
+
+    const depositId = `${transactionHash}_${event.index}`;
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [_, senderFelt, nostrRecipientFelt, starknetRecipientFelt] =
+      event.keys;
+
+    const sender = validateAndParseAddress(
+      `0x${FieldElement.toBigInt(senderFelt).toString(16)}`,
+    ) as ContractAddress;
+
+    const nostr_recipient = validateAndParseAddress(
+      `0x${FieldElement.toBigInt(nostrRecipientFelt).toString(16)}`,
+    ) as ContractAddress;
+
+    const starknet_recipient = validateAndParseAddress(
+      `0x${FieldElement.toBigInt(starknetRecipientFelt).toString(16)}`,
+    ) as ContractAddress;
+
+    const [
+      amountLow,
+      amountHigh,
+      tokenAddressFelt,
+      gasAmountLow,
+      gasAmountHigh,
+      gasTokenAddressFelt,
+    ] = event.data;
+
+    const amountRaw = uint256.uint256ToBN({
+      low: FieldElement.toBigInt(amountLow),
+      high: FieldElement.toBigInt(amountHigh),
+    });
+    const amount = formatUnits(amountRaw, constants.DECIMALS).toString();
+
+    const gasAmountRaw = uint256.uint256ToBN({
+      low: FieldElement.toBigInt(gasAmountLow),
+      high: FieldElement.toBigInt(gasAmountHigh),
+    });
+    const gasAmount = formatUnits(gasAmountRaw, constants.DECIMALS).toString();
+
+    const token_address = tokenAddressFelt
+      ? shortString.decodeShortString(
+          FieldElement.toBigInt(tokenAddressFelt).toString(),
+        )
+      : '';
+
+    const gas_token_address = gasTokenAddressFelt
+      ? shortString.decodeShortString(
+          FieldElement.toBigInt(gasTokenAddressFelt).toString(),
+        )
+      : '';
+
+    const data = {
+      network: 'starknet-sepolia',
+      blockHash,
+      blockNumber: Number(blockNumber),
+      blockTimestamp: new Date(Number(blockTimestamp.seconds) * 1000),
+      transactionHash,
+      depositId,
+      sender,
+      nostr_recipient,
+      starknet_recipient,
+      amount: Number(amount),
+      token_address,
+      gas_token_address,
+      gas_amount: Number(gasAmount),
+    };
+
+    await this.tipService.createClaim(data);
```

Comment:
Let's update the available of tip_deposit here:

boolean is_claimed to true and the gas amount